File: bin/Nodes/C/SVG.c
#include "Imogen.h"

typedef struct SVG_t
{
	char filename[1024];
	float dpi;
} SVG;

int main(SVG *param, Evaluation *evaluation, void *context)
{
	Image image;
	image.bits = 0;
	if (param->dpi <= 1.f)
		param->dpi = 96.f;

	if (strlen(param->filename))
	{
		if (LoadSVG(param->filename, &image, param->dpi) == EVAL_OK)
		{
			SetEvaluationImage(context, evaluation->targetIndex, &image);
		}
	}
	return EVAL_OK;
}

File: bin/Nodes/JS/SVG.js
function SVG(parameters, evaluation, context)
{
	image = new Image;
	if (parameters.dpi <= 1.0)
	{
		parameters.dpi = 96.0;
	}
	if (!(evaluation.dirtyFlag & DirtyParameter))
	{
		return EVAL_OK;
	}
	if (parameters.filename.length)
	{
		if (LoadSVG(parameters.filename, image, parameters.dpi) == EVAL_OK)
		{
			SetEvaluationImage(context, evaluation.targetIndex, image);
		}
	}
	return EVAL_OK;
}